### ISSUE TITLE
Recent books list is always empty

### ISSUE THREAD

**Issue by dagonalo** [seblucas/cops#501]
I have successfully configured COPS in my Synology NAS. Everything works fine except for the "recent books list" that always is blank. It used to work in the past (months ago) but one day stopped working.
I've upgraded to COPS 1.1.3 with same results.
Any suggestion on what could be the reason?

**Comment by marioscube** [comment]
All other menu options still work?

If it "one day stopped working" and you did not change anything, then I would think that it's a database problem.

To test, simply create a very small calibre database (5 books or so) and put this on your Synology and test it.
If it works: database problem. recreate your database in calibre and then sync with your Synology.
If not ....... let us know.

**Comment by dagonalo** [comment]
Yes, all other options work: searching books, authors, navigating through the pages, etc.
When this specific option stopped working (several months ago) I had not changed anything besides adding books regularly using the Calibre desktop software.
Anyway, I've recreated the database locally (not in a mapped network), then copied the metadata.db file back into the calibre NAS folder, and now the problem is solved.
Thank you!

**Comment by rjmcnamara** [comment]
TL;DR first: Check the date values of all books within Calibre (sort by date) and bulk-set them if they are missing.

I encountered this same behavior and dug into the code to find the query used to produce the recent additions page.   While attempting to simplify the query in DB Browser I got as far as :
`
select timestamp, comments.text from books left outer join comments on comments.book = books.id order by timestamp desc limit 10
`
...which failed with a malformed database error.   Removing 'order by timestamp desc'  clause allowed the query to work.  

The timestamps of about 50 books were using an '1101' year (I think) and within Calibre the "date" value for those books was unset.   Bulk setting the date fixed the problem entirely.
